src/gui/symbology/qgsinterpolatedlinesymbollayerwidget.cpp

```diff
@@ -107,10 +112,10 @@ QgsInterpolatedLineSymbolLayerWidget::QgsInterpolatedLineSymbolLayerWidget( QgsV
   connect( mColorEndFieldExpression, static_cast < void ( QgsFieldExpressionWidget::* )( const QString & ) >( &QgsFieldExpressionWidget::fieldChanged )
            , this, &QgsInterpolatedLineSymbolLayerWidget::reloadMinMaxColorFromLayer );
 
-  connect( mLineEditColorMinValue, &QLineEdit::textChanged, this, &QgsInterpolatedLineSymbolLayerWidget::onColorMinMaxLineTextChanged );
-  connect( mLineEditColorMinValue, &QLineEdit::textEdited, this, &QgsInterpolatedLineSymbolLayerWidget::onColorMinMaxLineTextEdited );
-  connect( mLineEditColorMaxValue, &QLineEdit::textChanged, this, &QgsInterpolatedLineSymbolLayerWidget::onColorMinMaxLineTextChanged );
-  connect( mLineEditColorMaxValue, &QLineEdit::textEdited, this, &QgsInterpolatedLineSymbolLayerWidget::onColorMinMaxLineTextEdited );
+  //connect( mLineEditColorMinValue, qOverload<double>( &QDoubleSpinBox::valueChanged ), this, &QgsInterpolatedLineSymbolLayerWidget::onColorMinMaxLineTextChanged );
```

Comment:
Can you remove the comments?

Reply:
I left it to have feedback whether I should remove the QgsInterpolatedLineSymbolLayerWidget::onColorMinMaxLineTextChanged method (line 350). So, remove?

Reply:
I see, it seems like the "changed" signal was meant to update the shader widget without triggering a reclassification while the "edited" signal was also triggering a reclassification, so it's better to leave it like it was before.

You'd have to check if the behavior remains the same we had before your changes.

Reply:
> You'd have to check if the behavior remains the same we had before your changes.

Yeah! This is what I did and led me to comment out those lines: I couldn't find when a simple update of widget was done without classification, and what would be the interest in doing so. When I had those lines enabled (assuming that my signal replacement is correct), I couldn't get the classification done automatically. 
Here a video from 3.22, whenever a color value change a classification is processed.

https://user-images.githubusercontent.com/7983394/153881153-31dfa854-621f-48a9-8b4f-4cfb4cdda3f8.mp4

@vcloarec Could you explain the rationale here, please? Thanks

Reply:
I think that the difference is that textChanged() is emited whenever the contents of the widget changes whereas textEdited() is emited only when the user changes the text using mouse and keyboard (so it is not emitted when you call ::setText()

Reply:
Yes, from what I can remember, that is the logic, to avoid reclassification if the value are not changed from the user. When it was implemented, I think it was the same logic as for scalar rendering of mesh layer, working with QLineEdit (as for raster ?).
 Since, in mesh layer, min/max are now handled by `QgsDoubleSpinBox` (thanks @elpaso ): https://github.com/qgis/QGIS/blob/ef85926a5d52ac918f2f0042f0487de0d66ce5bd/src/gui/mesh/qgsmeshrendererscalarsettingswidget.cpp#L52 
I guess the logic should be same,` textEdited` signals was removed, and `whileBlocking()` do the job.